### PR TITLE
Move MoveIt Config specific files out of bringup package

### PR DESCRIPTION
**Pull request by abake48**
Since the current state of `kortex_moveit.launch.py` is broken, now would be a good time to get the MoveIt config package cleaned up. 

This will clear up [this issue](https://github.com/PickNikRobotics/ros2_kortex/issues/33)
and is part of the work needed to [remove moveit ROS 1 packages](https://github.com/PickNikRobotics/ros2_kortex/issues/31)

### CODE REVIEW

File: kortex_moveit_config/package.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>kortex_moveit_config</name>
+  <version>2.0.0</version>
+  <description>
+     An example package with MoveIt 2 configurations for the Gen3 Kinova arms.
+  </description>
+  <author email="avannobel@kinova.ca">Alexandre Vannobel</author>
+  <maintainer email="avannobel@kinova.ca" />
```
Comment (moriarty) [inline kortex_moveit_config/package.xml]: I assume this is the original author of the ROS1 config but we should check what should go here for the ROS 2 version maintainer.

File: kortex2_bringup/launch/gen3.launch.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ def generate_launch_description():
             description="IP address by which the robot can be reached.",
         )
     )
+    declared_arguments.append(DeclareLaunchArgument("dof", description="DoF of robot."))
```
Comment (MarqRazz) [inline kortex2_bringup/launch/gen3.launch.py]: Can you add a default_value here? I think it should load the Gen3 if the user does not specify.

File: kortex2_bringup/launch/gen3.launch.py [hunk 3]
```diff
@@ -51,21 +52,47 @@ def generate_launch_description():
...
+    gripper = LaunchConfiguration("gripper")
+    use_internal_bus_gripper_comm = LaunchConfiguration("use_internal_bus_gripper_comm")
+    launch_rviz = LaunchConfiguration("launch_rviz")
 
     base_launch = IncludeLaunchDescription(
-        PythonLaunchDescriptionSource([ThisLaunchFileDir(), "/ur_control.launch.py"]),
+        PythonLaunchDescriptionSource([ThisLaunchFileDir(), "/kortex_control.launch.py"]),
         launch_arguments={
-            "ur_type": "gen3",
+            "robot_type": "gen3",
```
Comment (MarqRazz) [inline kortex2_bringup/launch/gen3.launch.py:96]: This should be a launch argument.

File: kortex2_bringup/launch/kortex_control.launch.py [hunk 4]
```diff
@@ -305,7 +306,7 @@ def generate_launch_description():
     robot_pos_controller_spawner = Node(
         package="controller_manager",
         executable="spawner",
-        arguments=[robot_pos_controller, "--stopped", "-c", "/controller_manager"],
+        arguments=[robot_pos_controller, "-c", "/controller_manager"],
     )
```
Comment (MarqRazz) [inline kortex2_bringup/launch/kortex_control.launch.py]: This controller can not be started while the `JTC` is running. The hardware only allows a single command mode be active.

File: kortex_moveit_config/config/kortex_servo.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
+###############################################
+# Modify all parameters related to servoing here
+###############################################
+use_gazebo: false # Whether the robot is started in a Gazebo simulation environment
```
Comment (MarqRazz) [inline kortex_moveit_config/config/kortex_servo.yaml]: I recommend you add some comments to this file that the hardware does not perform well with Servo (we recommend you use the `twist_controller` interface on the robot) and this should only be used for simulation.

**Review by MarqRazz — COMMENTED** [review]
Are you planning to run MSA to create the `moveit_config` packages? It uses the `MoveItConfigsBuilder` to give the users an easy way to simulate the hardware.

File: kortex_moveit_config/package.xml (hunk 1 above)
Comment (abake48) [reply kortex_moveit_config/package.xml]: I'll ask about this today in standup. I wasn't sure if I should change them but it makes sense in my mind to

File: kortex2_bringup/launch/gen3.launch.py (hunk 3 above)
Comment (abake48) [reply kortex2_bringup/launch/gen3.launch.py:96]: I made this launch file specifically for launching the gen 3 arm. It was an attempt at mirroring the [UR Bringup package](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/tree/main/ur_bringup/launch). Would you advise against doing so? I wasn't sure if I should create a bringup package for each arm or just have launch files for each.

**Comment by abake48** [comment]
> Are you planning to run MSA to create the `moveit_config` packages? It uses the `MoveItConfigsBuilder` to give the users an easy way to simulate the hardware.

I wasn't initially but this would be a good place to actually use MSA and build out/report missing features. I'll give it a shot

File: kortex2_bringup/launch/gen3.launch.py (hunk 3 above)
Comment (MarqRazz) [reply kortex2_bringup/launch/gen3.launch.py:96]: Sorry I missed that this was specifically for the Gen3. This is fine, but think the default should also include a working gripper.

**Comment by MarqRazz** [comment]
> I wasn't initially but this would be a good place to actually use MSA and build out/report missing features. I'll give it a shot

Let me know if you run into any issues. We might want to get the robotiq swapped in before we make the moveit_configs (I will try and get to this tomorrow morning).

**Comment by abake48** [comment]
> Let me know if you run into any issues.

Ran into an issue right off the bat: turns out the Gen 3 7dof urdfs were out of date so I made a [PR](https://github.com/PickNikRobotics/ros2_kortex/pull/94) to get those updated and working with MSA

File: kortex_moveit_config/CMakeLists.txt
Comment (MarqRazz) [inline kortex_moveit_config/CMakeLists.txt:1]: Why did you add this file and then add a COLCON_IGNORE to the folder?

**Review by MarqRazz — COMMENTED** [review]
Looks like you still have a bunch of old launch files in some of the moveit_configs. What is the goal here? Just to rename the folder `kotrex_move_it_config`?

File: kortex_moveit_config/CMakeLists.txt
Comment (abake48) [reply kortex_moveit_config/CMakeLists.txt:1]: I was working on updating the MoveIt configs in this PR but have decided to make a separate PR for that. I initially tried to have the moveit config as a single package but I now see that that is a bad idea. The CMakeLists.txt file and COLCON_IGNORE are artifacts of that. I'll clean this up now

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -81,7 +81,7 @@ repos:
         name: ament_cppcheck
         description: Static code analysis of C/C++ files.
         stages: [commit]
-        entry: ament_cppcheck
+        entry: env AMENT_CPPCHECK_ALLOW_SLOW_VERSIONS=1 ament_cppcheck
```
Comment (moriarty) [inline .pre-commit-config.yaml:85]:
```suggestion
        # enable slow version of cpp check: https://github.com/ament/ament_lint/pull/345
        entry: env AMENT_CPPCHECK_ALLOW_SLOW_VERSIONS=1 ament_cppcheck
```

File: kortex_moveit_config/readme.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<!--
+* KINOVA (R) KORTEX (TM)
+*
+* Copyright (c) 2018 Kinova inc. All rights reserved.
```
Comment (moriarty) [inline kortex_moveit_config/readme.md:4]: GitHub says this is a new file, are you moving it from somewhere?

File: kortex2_bringup/package.xml [hunk 8]
```diff
@@ -2,7 +2,7 @@
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>kortex2_bringup</name>
-  <version>0.0.0</version>
+  <version>1.0.0</version>
```
Comment (moriarty) [inline kortex2_bringup/package.xml]:
```suggestion
  <version>0.0.1</version>
```

Are we ready to call this 1.0.0 or should we call it 0.0.1 and then bump it when we actually make the public launch?

**Review by moriarty — CHANGES_REQUESTED** [review]
The only change I would request is the version number because we'll probably want to release 1.0.0 when we actually launch, and just bump all versions at once.

But It's not a strong blocking request and I could be convinced otherwise.

I left a question about the 2018 copywrite on a new file, I assume GitHub just isn't able to track that file is just being moved.

And I would add the comment documenting what the environment variable does with cppcheck

File: kortex_moveit_config/readme.md (hunk 7 above)
Comment (abake48) [reply kortex_moveit_config/readme.md:4]: I moved it from `kinova_move_it_config`. Unsure why it says it's a new file

File: kortex2_bringup/package.xml (hunk 8 above)
Comment (abake48) [reply kortex2_bringup/package.xml]: I was thinking we should bump it on public release.

**Review by moriarty — APPROVED** [review]

**Pull request merged**